libs/ui/controls/TagSelectPicker.js

```diff
@@ -0,0 +1,147 @@
+import { html, useState, useEffect } from '../../deps/htm-preact.js';
+import useDebounce from '../../hooks/useDebounce.js';
+
+const Tag = ({
+  id,
+  label,
+  hasChildren,
+  isChecked,
+  onCheck,
+  onExpand,
+}) => {
+  return html`
+    <div
+      class="tagselect-item"
+      key=${id}
+      data-key=${id}
+      onClick=${hasChildren ? onExpand : onCheck}
+    >
+      <input id=${id} type="checkbox" class="cb ${isChecked ? 'checked' : ''}" />
+      <label class="label">${label?.replace('&amp;', '&')}</label>
+      ${hasChildren ? html`<span class="has-children"></span>` : ''}
```

Comment:
Would this make more sense as a button like in the UI Accordion?
```js
<button aria-label=${buttonText}></button>
```

Reply:
It was built as a `span` for the caas configurator so I kept it that way. It's probably built that way because the action isn't actually attached to the span, it's on the div.